src/Features/Core/Portable/SemanticSearch/AbstractSemanticSearchService.cs

```diff
@@ -207,7 +236,7 @@ private static bool TryGetFindMethod(Assembly queryAssembly, [NotNullWhen(true)]
         {
             try
             {
-                (method, queryKind) = GetFindMethod(program, ref error);
+                (method, queryKind, isAsync) = GetFindMethod(program, ref error);
```

Comment:
so... looking at how this is used, i have an alternate suggestion.  instead of returning this isAsync flag, just always return an IAsyncEnumerable.  If you get an IEnumerable, then that is *trivially* wrappable in an IAsyncEnumerable.  Which then means all subsequent code operates uniformly.

Reply:
Good idea